server/src/main/scala/actors/PortStateActor.scala

```diff
@@ -66,23 +91,89 @@ class PortStateActor(liveStateActor: AskableActorRef,
       val updates: Option[PortStateUpdates] = state.updates(millis, start, end)
       sender() ! updates
 
+    case GetFlights(startMillis, endMillis) =>
+      val start = SDate(startMillis)
+      val end = SDate(endMillis)
+      log.info(s"Got request for flights between ${start.toISOString()} - ${end.toISOString()}")
+      val flightsToSend = state.flights.range(start, end).values.toList
+      sender() ! FlightsWithSplits(flightsToSend, List())
+
     case unexpected => log.warn(s"Got unexpected: $unexpected")
   }
 
   def stateForPeriod(start: MillisSinceEpoch, end: MillisSinceEpoch): Option[PortState] = Option(state.window(SDate(start), SDate(end)))
 
   def stateForPeriodForTerminal(start: MillisSinceEpoch, end: MillisSinceEpoch, terminalName: TerminalName): Option[PortState] = Option(state.windowWithTerminalFilter(SDate(start), SDate(end), Seq(terminalName)))
 
-  def splitDiffAndSend(diff: PortStateDiff): Unit = {
+  val flightMinutesBuffer: mutable.Set[MillisSinceEpoch] = mutable.Set[MillisSinceEpoch]()
+  val loadMinutesBuffer: mutable.Map[TQM, LoadMinute] = mutable.Map[TQM, LoadMinute]()
+
+  def splitDiffAndSend(diff: PortStateDiff, uuid: String): Unit = {
     val replyTo = sender()
 
+    log.debug(s"Processing incoming PortStateMinutes: $uuid")
+
+    if (diff.flightMinuteUpdates.nonEmpty) flightMinutesBuffer ++= diff.flightMinuteUpdates
+    if (diff.crunchMinuteUpdates.nonEmpty) loadMinutesBuffer ++= crunchMinutesToLoads(diff).map(lm => (lm.uniqueId, lm))
+
+    handleCrunchRequest()
+    handleSimulationRequest()
+
     splitDiff(diff) match {
       case (live, forecast) =>
-        for {
-          _ <- liveStateActor.ask(live)
-          _ <- forecastStateActor.ask(forecast)
-        } yield replyTo ! Ack
+        val asks = List(
+          (liveStateActor, live) -> "live crunch persistence request failed",
```

Comment:
this is not very clear.

Reply:
It sure isn't. I've refactored and hopefully made it clearer now